.github/workflows/release.yml

```diff
@@ -0,0 +1,27 @@
+name: Release to PyPI
+on:
+  push:
+    tags: ["*"]
+
+jobs:
+  release:
+    runs-on: ubuntu-22.04
+    environment:
+      name: release
+      url: https://pypi.org/p/pyproject-fmt
+    permissions:
+      id-token: write
+    steps:
+      - name: Setup python to build package
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Install build
+        run: python -m pip install build
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Build package
+        run: pyproject-build -s -w . -o dist
```

Comment:
@gaborbernat FYI this place is an attack surface targeting OIDC as it can potentially be used to reach other things than PyPI. The attack vector is the build deps poisoning, followed by the privilege elevation. This is why I always recommend building in a separate job that does not have access to OIDC.

Reply:
I don't see how that helps. The token is scoped to this project only. If the user has hijacked the build chain, they cannot upload a new package themselves, however a package upload will happen anyway as part of the release pipeline, so they can just alter the build output and, in essence, upload whatever they want.